src/app/ReadClient.cpp

```diff
@@ -70,6 +126,16 @@ ReadClient::~ReadClient()
 
 void ReadClient::Close(CHIP_ERROR aError)
 {
+    if (aError != CHIP_NO_ERROR)
```

Comment:
Why did we move this logic up from where it was originally?

Reply:
i think it is fine to keep it in original location, just put it back, thanks